types/stripe/index.d.ts

```diff
@@ -6790,7 +6812,7 @@ declare namespace Stripe {
      * A set of key/value pairs that you can attach to an object. It can be useful for storing
      * additional information about the object in a structured format.
      */
-    interface IMetadata { }
+    type IMetadata = any;
```

Comment:
Why is this necessary?

Reply:
Because of my proposed test case: [Line 248](https://github.com/CodeAnimal/DefinitelyTyped/blob/3dcc65aa22e52475e64cbe97742186f68615f83c/types/stripe/stripe-tests.ts#L248)

I agree, that `any` may not be the best type, but working with the metadata field can feel awkward and clumsy with the interface model, as the test example displays. It seems like having to jump through hoops, needing to cast the metadata field every time I want to read data from it. I'm not sure what the best solution is for this, what do we think?

Reply:
@LinusU @andy-ms Maybe this would work better:

```typescript
interface IMetadata {
    [x: string]: string | number;
}
```

This would cover all of these test cases:
```typescript
customer.metadata["test"] = "test";
customer.metadata["test"] = 1;
customer.metadata["test"] == "test";
customer.metadata["test"] == 1;
customer.metadata.test = "test";
customer.metadata.test = 1;
customer.metadata = {
    test: "test"
}
customer.metadata = null;
```

But wouldn't allow:
```typescript
customer.metadata = true;
customer.metadata = "test";
customer.metadata = 2;
customer.metadata.test = { nested: "Pojo"};
customer.metadata = {
    test: { nested: "Pojo" }
}
```

You would need to set the `TypeScript Version` header to 2.2 or above though. What do we think?

Reply:
That seems very good!

Are you sure that you are allowed to send numbers to Stripe though? Or should it just be `{ [key: string]: string }`?

Reply:
I think the stripe-node library just stringifies the value to a url string, using the __qs__ package (see [here](https://github.com/stripe/stripe-node/blob/ab9fbfb72511afabfdb6b25ebe0ba3d8fef626e3/lib/StripeResource.js#L247) and [here](https://github.com/stripe/stripe-node/blob/ab9fbfb72511afabfdb6b25ebe0ba3d8fef626e3/lib/utils.js#L29)).

For example:
```typescript
metadata: {
    test: "123"
}
```
and
```typescript
metadata: {
    test: 123
}
```
should both stringify down to the url value: `"metadata[test]=123"` (at least the [docs](https://www.npmjs.com/package/qs#stringifying) seem to indicate that). You can see that in the [Stripe Metadata docs](https://stripe.com/docs/api/curl#metadata), if you compare the code examples for Node and cURL. Either way, I'm 99.999% confident that qs.stringify would treat the `123` and `"123"` as the same with regards to the output.

With that being the case, it seems valid to allow both a `number` or a `string`, as both are equally as valid.

Reply:
You might have a point on the __return value__ though. It will always return a string, maybe that should be taken into consideration.